### PR TITLE
7.0 bundle import base

### PR DESCRIPTION
**Pull request by sebastienbeau**
Migrate this merge proposal from lp : https://code.launchpad.net/~akretion-team/openerp-connector-magento/7.0-next-release-bundle-import-base/+merge/223211

This Pull Request depends of pull requests:
- [ ] OCA/stock-logistics-workflow#101 
- [ ] OCA/account-invoicing#69

### CODE REVIEW

**Comment by chafique-delli** [comment]
This Pull Request depends of  pull requests 'https://github.com/OCA/sale-workflow/pull/138' and 'https://github.com/OCA/sale-workflow/pull/139'.

Best Regards,

**Comment by chafique-delli** [comment]
This Pull Request depends  of pull requests https://github.com/OCA/stock-logistics-workflow/pull/101  and https://github.com/OCA/account-invoicing/pull/69 .

Best Regards,

**Comment by chafique-delli** [comment]
Hi, @guewen, can you review my pull request.
Thanks,

File: magentoerpconnect_bundle_split/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ 'category': 'Connector',
+ 'depends': ['magentoerpconnect',
+             'picking_line_sale_link',
+             'invoice_line_sale_link',
+             ],
+ 'author': 'MagentoERPconnect Core Editors',
+ 'license': 'AGPL-3',
+ 'description': """
+Magento Connector - BUNDLE SPLIT
+=======================================
```
Comment (guewen) [inline magentoerpconnect_bundle_split/__openerp__.py]: Underline is too long. And can you remove caps?

File: magentoerpconnect_bundle_split/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+##############################################################################
+
+{'name': 'Magentoerpconnect Bundle Split',
+ 'version': '1.0.0',
+ 'category': 'Connector',
+ 'depends': ['magentoerpconnect',
+             'picking_line_sale_link',
+             'invoice_line_sale_link',
+             ],
+ 'author': 'MagentoERPconnect Core Editors',
```
Comment (guewen) [inline magentoerpconnect_bundle_split/__openerp__.py]: Could you put 'Akretion,Odoo Community Association (OCA)' in there?

File: magentoerpconnect_bundle_split/sale.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+class SaleOrderLineBundleImportMapper(sale.SaleOrderLineBundleImportMapper):
+    _model_name = 'magento.sale.order.line'
+
+    def price_is_zero(self, record):
+        product_binder = self.get_binder_for_model('magento.product.product')
+        magento_product_id = product_binder.to_openerp(record['product_id'])
+        magento_product = self.session.browse('magento.product.product',
+                                              magento_product_id)
+        if magento_product.price_type == 'dynamic':
+            return True
```
Comment (guewen) [inline magentoerpconnect_bundle_split/sale.py]: just `return`

File: magentoerpconnect/sale.py [hunk 4]
```diff
@@ -1024,6 +1047,14 @@ class MagentoSaleOrderOnChange(SaleOrderOnChange):
 
 
 @magento
+class SaleOrderLineBundleImportMapper(ConnectorUnit):
```
Comment (guewen) [inline magentoerpconnect/sale.py]: It is not a Mapper, could you rename it BundleLineStrategy?

File: magentoerpconnect/sale.py [hunk 5]
```diff
@@ -482,6 +483,14 @@ def move(self, binding):
 
 
 @magento
+class SaleOrderBundleImport(ConnectorUnit):
```
Comment (guewen) [inline magentoerpconnect/sale.py]: It is named as if it was an Importer but it is not. Could you rename it to BundleLineLinker?

File: magentoerpconnect/sale.py [hunk 6]
```diff
@@ -631,7 +646,15 @@ def _link_parent_orders(self, binding_id):
                                    {'canceled_in_backend': True})
             current_bind_id = parent_bind_id
 
+    def _link_hierarchical_lines(self, binding_id):
```
Comment (guewen) [inline magentoerpconnect/sale.py]: I think you could simplify the name to `_link_lines`.

File: magentoerpconnect/sale.py [hunk 7]
```diff
@@ -235,6 +235,7 @@ def _get_lines_from_order(self, cr, uid, ids, context=None):
                                  digits_compute=dp.get_precision('Account')),
         # XXX common to all ecom sale orders
         'notes': fields.char('Notes'),
+        'magento_parent_item_id': fields.integer('Magento Parent Item ID'),
```
Comment (guewen) [inline magentoerpconnect/sale.py]: To remove if possible (see below)

File: magentoerpconnect_bundle_split/sale.py [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+@magento(replacing=sale.SaleOrderBundleImport)
+class SaleOrderBundleImport(sale.SaleOrderBundleImport):
+    _model_name = ['magento.sale.order']
+
+    def link_hierarchical_lines(self, binding_id):
+        line_binder = self.get_binder_for_model('magento.sale.order.line')
+        magento_sale_order = self.session.browse(self.model._name, binding_id)
+        for line in magento_sale_order.magento_order_line_ids:
+            if line.magento_parent_item_id:
```
Comment (guewen) [inline magentoerpconnect_bundle_split/sale.py]: `magento_parent_item_id` is stored solely for this purpose?
I suggest to remove this field and to pass the magento data as an argument to this method so the parent lines can be found in the magento data.

**Comment by guewen** [comment]
Added some suggestions.
Also, could you clean the diff? There is a lot of commit that shouldn't be there.

**Comment by guewen** [comment]
@sebastienbeau any news?

**Pull request closed without merging**